# Towards reproducible analysis of benthos structural complexity: A case study on Antarctic polychaete reefs using action cameras and remotely operated vehicles
- Aim: Relationships between substrate type and structural complexity
- Goal: Create a complete data pipeline for data analysis and visualization
- Authors: J.C. Montes-Herrera, G. Johnstone, J. Stark, N. Hill, V. Cummings, V. Lucieer
- Contact: juancarlos.montesherrera@utas.edu.au
- Associated publication submitted to journal Remote Sensing in Ecology & Conservation.

## Step 2 - Structural complexity metrics from Fukunaga et al. (2019) 
- Once quadrats are generated and stored in "data\quadrats" they are processed to quantify structural complexity metrics
- Metrics were generated using Fukunaga et al. (2019) code https://doi.org/10.3390/jmse7020027
- Fukunaga et al. (2019) code used in this study: ant_biogenic_structures\code\Fukunaga_2019-CodeCompiled.R
- Note that this step could be performed in the same notebook using the rpy2 module (https://rpy2.github.io/).
- Resulting file '\ant_biogenic_structures\data\quadrats\habitat-metrics-16-32.csv'

In [1]:
import numpy as np
import pandas as pd

In [2]:
metrics_file = r"C:\Users\jcmontes\Documents\GitHub\ant_biogenic_structures\data\habitat-metrics-16-32.csv"
df = pd.read_csv(metrics_file).set_index('file_name')
df.head()

orig_area  surface16  planerS16  surface32  planerS32  \
file_name                                                                
A-DEM__1_00.tif     1.0300   5.821141     0.9216   5.821141     0.9216   
A-DEM__1_01.tif     1.0300   2.728238     0.9216   2.728238     0.9216   
A-DEM__1_10.tif     1.0403   8.701784     0.9216   8.701784     0.9216   
A-DEM__1_11.tif     1.0403   6.659451     0.9216   6.659451     0.9216   
A-DEM__2_00.tif     1.0932   2.585702     0.8448   1.759163     0.5120   

                     max_h      min_h      fd16      fd32  surface_complexity  \
file_name                                                                       
A-DEM__1_00.tif  30.359108  25.954962  2.224226  2.217311            6.316342   
A-DEM__1_01.tif  31.765301  30.216726  2.194972  2.201625            2.960328   
A-DEM__1_10.tif  30.384106  25.770912  2.308227  2.266483            9.442040   
A-DEM__1_11.tif  31.162127  26.774553  2.171841  2.174548            7.225967   
A-DEM__2_00.tif  20.271091  19.179802  2.295925  2.305882            3.060726   

                 mean_slope  mean_profile_curvature  mean_plan_curvature  
file_name                                                                 
A-DEM__1_00.tif   61.009223               45.803822            35.273526  
A-DEM__1_01.tif   47.855155                4.697123             2.477279  
A-DEM__1_10.tif   48.212245              108.556322           101.512511  
A-DEM__1_11.tif   60.925839               24.957675            27.305298  
A-DEM__2_00.tif   51.362519                9.367260             7.587447

In [3]:
## Adjust index to make it prettier
df.index = df.index.str.replace(r'.tif$', '', regex=True); df.index = df.index.str.replace(r'DEM__', '', regex=True)
df.index = df.index.str.replace(r'DEM_', '', regex=True);
structuralcomplexity = df
structuralcomplexity.head()

orig_area  surface16  planerS16  surface32  planerS32      max_h  \
file_name                                                                     
A-1_00        1.0300   5.821141     0.9216   5.821141     0.9216  30.359108   
A-1_01        1.0300   2.728238     0.9216   2.728238     0.9216  31.765301   
A-1_10        1.0403   8.701784     0.9216   8.701784     0.9216  30.384106   
A-1_11        1.0403   6.659451     0.9216   6.659451     0.9216  31.162127   
A-2_00        1.0932   2.585702     0.8448   1.759163     0.5120  20.271091   

               min_h      fd16      fd32  surface_complexity  mean_slope  \
file_name                                                                  
A-1_00     25.954962  2.224226  2.217311            6.316342   61.009223   
A-1_01     30.216726  2.194972  2.201625            2.960328   47.855155   
A-1_10     25.770912  2.308227  2.266483            9.442040   48.212245   
A-1_11     26.774553  2.171841  2.174548            7.225967   60.925839   
A-2_00     19.179802  2.295925  2.305882            3.060726   51.362519   

           mean_profile_curvature  mean_plan_curvature  
file_name                                               
A-1_00                  45.803822            35.273526  
A-1_01                   4.697123             2.477279  
A-1_10                 108.556322           101.512511  
A-1_11                  24.957675            27.305298  
A-2_00                   9.367260             7.587447

In [4]:
structuralcomplexity_clean = structuralcomplexity.drop(['A-1_00','A-1_10','A-1_11','G-2_00', 'G-2_01'])

In [5]:
structuralcomplexity_clean.shape

(46, 13)

In [6]:
#Export
# structuralcomplexity_clean.to_csv(r"C:\Users\jcmontes\Documents\GitHub\ant_biogenic_structures\data\structural_metrics_clean.csv")

### Fractal dimension values - Table for reviewers

In [7]:
## Add Transect column
transect = []

for i, quadrat in enumerate(structuralcomplexity_clean.index):
    t = structuralcomplexity_clean.index[i][0]
    transect.append(t);

structuralcomplexity_clean.insert(loc=0, column = 'Transect', value = np.array(transect))

In [8]:
t_structural = structuralcomplexity_clean.groupby(['Transect'])

In [14]:
t_structural.mean()

orig_area  surface16  planerS16  surface32  planerS32      max_h  \
Transect                                                                     
A          1.072863   2.630230   0.817600   2.144585   0.652800  19.583071   
B          1.055200   1.605119   0.921600   1.605119   0.921600   8.362888   
C          1.277167   1.838847   1.083733   1.621705   1.024000   0.539816   
D          1.129575   3.373921   0.896000   2.999721   0.793600   7.028465   
E          1.102100   2.626818   0.853333   2.439034   0.785067  -0.867987   
F          1.103883   1.821781   0.861867   1.667901   0.802133  -1.292530   
G          1.093800   2.443664   0.887467   2.370820   0.853333  -0.824978   
H          1.097743   1.681837   0.833829   1.422004   0.716800  -0.976449   

              min_h      fd16      fd32  surface_complexity  mean_slope  \
Transect                                                                  
A         18.361725  2.280123  2.264270            3.223612   50.470831   
B          7.618711  2.122250  2.109268            1.741665   31.600329   
C         -0.466882  2.114393  2.087253            1.696193   38.411115   
D          5.462011  2.310473  2.287210            3.779028   54.409474   
E         -1.919073  2.332400  2.298668            3.078732   53.125920   
F         -2.981506  2.082165  2.095079            2.113727   51.424798   
G         -2.322224  2.152090  2.159923            2.722008   48.139089   
H         -1.641656  2.177940  2.160715            2.006147   30.868129   

          mean_profile_curvature  mean_plan_curvature  
Transect                                               
A                       6.206636             5.490236  
B                      -6.966234            -5.919929  
C                      -4.314897            -3.303509  
D                       7.316948             7.554377  
E                       4.767906             4.582915  
F                      -1.034793            -0.276884  
G                       0.797398             0.004410  
H                      -6.178411            -5.709895

In [15]:
t_mean = t_structural[['fd16','fd32','surface_complexity','mean_slope','mean_profile_curvature','mean_plan_curvature']].mean()
t_std = t_structural[['fd16','fd32','surface_complexity','mean_slope','mean_profile_curvature','mean_plan_curvature']].std()

In [17]:
t_mean.to_markdown(r"C:\Users\jcmontes\Desktop\mean.md")

In [19]:
t_std.to_markdown(r"C:\Users\jcmontes\Desktop\std.md")

Steps:
- Open in markdown inside Jupyter and use multicursor for column pastng, could try multiple columns.
- Save as markdown
- Export to docx via pandoc

| Transect   |    fd16 |    fd32 |   surface_complexity |   mean_slope |   mean_profile_curvature |   mean_plan_curvature |
|:-----------|--------:|--------:|---------------------:|-------------:|-------------------------:|----------------------:|
| A          | 2.28(±0.09) | 2.26(±0.07) |              3.22(±0.77) |      50.47(±5.37)|                 6.20(±6.27)  |            5.49(±6.19)    |
| B          | 2.12(±0.09) | 2.10(±0.08) |              1.74(±0.83) |      31.60(±9.24)|                -6.96(±6.32)  |           -5.91(±6.20)    |
| C          | 2.11(±0.01) | 2.08(±0.03) |              1.69(±0.01) |      38.41(±0.51)|                -4.31(±1.81)  |           -3.30(±2.05)    |
| D          | 2.31(±0.08) | 2.28(±0.07) |              3.77(±0.83) |      54.40(±3.60)|                 7.31(±5.16)  |            7.55(±4.14)    |
| E          | 2.33(±0.02) | 2.29(±0.01) |              3.07(±0.21) |      53.12(±1.75)|                 4.76(±2.46)  |            4.58(±1.29)    |
| F          | 2.08(±0.02) | 2.09(±0.02) |              2.11(±0.26) |      51.42(±7.82)|                -1.03(±1.47)  |           -0.27(±0.70)   |
| G          | 2.15(±0.04) | 2.15(±0.04) |              2.72(±0.90) |      48.13(±4.51)|                 0.79(±3.50) |            0.004(±4.26) |
| H          | 2.17(±0.10) | 2.16(±0.09) |              2.00(±0.97) |      30.86(±9.61)|                -6.17(±2.86)  |           -5.70(±2.60)    |

In [ ]:
| Transect   |      fd16 |      fd32 |   surface_complexity |   mean_slope |   mean_profile_curvature |   mean_plan_curvature |
|:-----------|----------:|----------:|---------------------:|-------------:|-------------------------:|----------------------:|
| A          | 0.0923262 | 0.078664  |            0.771452  |     5.37755  |                  6.2756  |              6.19272  |
| B          | 0.0978161 | 0.0893964 |            0.831736  |     9.24704  |                  6.32584 |              6.20728  |
| C          | 0.0109467 | 0.0371235 |            0.0176297 |     0.512022 |                  1.81475 |              2.05333  |
| D          | 0.0821466 | 0.0748387 |            0.83401   |     3.60427  |                  5.1641  |              4.14813  |
| E          | 0.0208653 | 0.0158122 |            0.214452  |     1.75312  |                  2.46032 |              1.2947   |
| F          | 0.020762  | 0.020949  |            0.26016   |     7.82538  |                  1.47329 |              0.706123 |
| G          | 0.040152  | 0.0419956 |            0.901086  |     4.51701  |                  3.50531 |              4.26831  |
| H          | 0.109697  | 0.0948391 |            0.9771    |     9.61567  |                  2.86461 |              2.60123  |